packages/coinjoin/src/client/CoinjoinClient.ts

```diff
@@ -140,10 +148,30 @@ export class CoinjoinClient extends EventEmitter {
         this.onStatusUpdate({ rounds, changed });
     }
 
+    // emit log events to wallet
+    private log(message: string) {
+        if (this.listenerCount('log') < 1) return;
+        // redact log messages, parts wrapped into <...> like accountKey, round id, outpoint etc
+        const redacted = message.replace(/<(.*)?>}/g, match => {
+            if (match.length > 16) {
+                return `${match.substring(1, 9)}...${match.substring(
+                    match.length - 9,
+                    match.length - 1,
+                )}`;
+            }
+            return `[redacted]`;
+        });
+
+        this.emit('log', redacted);
```

Comment:
not important for this PR, but what about this?
```suggestion
        this.emit('log', isDevEnv ? message : redacted);
```
Or what is the point of putting those data in logs if they are always redacted?

Reply:
you can still see first/last chars of xpub/address from redacted message so its enough for debugging but not enough to consider it as sensitive data